### PR TITLE
Initial solution

### PR DESCRIPTION
**Pull request by DashaPoliazz**
[DEMO LINK](https://DashaPoliazz.github.io/react_movies-list-fetch-movies/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+import './App.scss';
+
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = (movieToAdd: Movie) => {
+    const isMovie = movies.find(movieToFind => {
+      return movieToFind.imdbId === movieToAdd.imdbId;
+    }) ? true : false;
```
Comment (solaryasha) [inline src/App.tsx:14]: use method `some` instead

File: src/App.tsx [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
+import './App.scss';
+
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = (movieToAdd: Movie) => {
```
Comment (solaryasha) [inline src/App.tsx:11]: Use `useCallback` for rendering optimization

File: src/components/FindMovie/FindMovie.tsx [hunk 3]
```diff
@@ -1,10 +1,67 @@
...
+import classNames from 'classnames';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+type Props = {
+  addMovie: (movieToAdd: Movie) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
```
Comment (solaryasha) [inline src/components/FindMovie/FindMovie.tsx:14]: use React.memo

File: src/components/FindMovie/FindMovie.tsx [hunk 4]
```diff
@@ -1,10 +1,67 @@
...
+  const [isLoading, setIsLoading] = useState(false);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    getMovie(query)
+      .then(res => {
```
Comment (solaryasha) [inline src/components/FindMovie/FindMovie.tsx:25]:
```suggestion
      .then(response => {
```

**Review by solaryasha — CHANGES_REQUESTED** [review]

File: src/api.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { MovieData } from './types/MovieData';
 import { ResponseError } from './types/ReponseError';
 
-const API_URL = 'https://www.omdbapi.com/?apikey=your-key';
+const API_URL = 'http://www.omdbapi.com/?i=tt3896198&apikey=81ae11e5';
```
Comment (VETALEK) [inline src/api.ts:4]:
```suggestion
const API_URL = 'https://www.omdbapi.com/?i=tt3896198&apikey=81ae11e5';
```
image

**Review by VETALEK — CHANGES_REQUESTED** [review]
GJ 🔥 
use HTTPS 😉

**Pull request closed without merging**